pkg/grub/grub_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package grub
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestFields(t *testing.T) {
```

Comment:
does grub itself have some unit tests for this stuff that we could copy over, too? that'd give me the most confidence that this is correct

Reply:
Well, grub does have some tests...
Most of then compare the output of grub to the output of bash on some code using mostly `echo`.
But they also use variables that I did not support at all...
I could probably extract the strings from the grub test and add them in the test case with comment saying where it comes from.
But I don't know what to do with the variables.

Reply:
And the test files are GPL licensed ... I don't think I can include them as testdata...

Reply:
Also as it's using `echo` we are loosing the tokens... ie we got `string` not `[]string`...

Reply:
mostly done. To go further we need to handle variables. The good thing is that for testing it we just need to import more tests from grub sources !